### ISSUE TITLE
missing enableGeoIndexing support in replace methods

### ISSUE THREAD

**Issue by sylvaindumont** [strongloop/loopback-connector-mongodb#472]
# Description

replace methods aren't saving `GeoPoint` with correct format with `"enableGeoIndexing": true`

enableGeoIndexing was implemented in 8112ebc720c4dbb74b72db3e44260905ed25128c, before
replace methods in 1fce12ec75fca9e3542c27d65cf6e1421037303b

geoIndexing support seems to have been forgotten during replace methods implementation.

# Link to reproduction sandbox

* set `"enableGeoIndexing": true` in mongodb datasource configuration
* configure a model with a property of `GeoPoint` type
* `create` a new instance of this model with data in the geopoint property
* update this instance with `replaceById`
* get this instance with `findById`

repro https://github.com/sylvaindumont/loopback-sandbox/
repro commit https://github.com/sylvaindumont/loopback-sandbox/commit/849d0fcfbe1b59c4b0efe6b68ff6d2bc234fe1ce

# Expected result

GeoPoint property should be saved with format
```json
{
  "coordinates": [
    5.513841,
    47.101878
  ],
  "type": "Point"
}
```
## Actual result
GeoPoint property is saved with format
```json
{
  "lat": 47.101878,
  "lng": 5.513841
}
```
on read operations https://github.com/strongloop/loopback-connector-mongodb/blob/master/lib/mongodb.js#L389 try to access `coordinates[0]` and crash the whole app

# Additional information

```
TypeError: Cannot read property '1' of undefined
    at MongoDB.fromDatabase (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/lib/mongodb.js:389:33)
    at /home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/lib/mongodb.js:1338:18
    at Array.map (<anonymous>)
    at /home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/lib/mongodb.js:1329:23
    at result (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb/lib/utils.js:414:17)
    at executeCallback (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb/lib/utils.js:406:9)
    at handleCallback (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb/lib/utils.js:128:55)
    at cursor.close (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb/lib/operations/cursor_ops.js:211:62)
    at handleCallback (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb/lib/utils.js:128:55)
    at completeClose (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb/lib/cursor.js:893:14)
    at _endSession (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb/lib/cursor.js:904:37)
    at Cursor._endSession (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb-core/lib/cursor.js:194:5)
    at Cursor._endSession (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb/lib/cursor.js:226:59)
    at Cursor.close (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb/lib/cursor.js:904:19)
    at cursor._next (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb/lib/operations/cursor_ops.js:211:23)
    at handleCallback (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb-core/lib/cursor.js:203:5)
    at _setCursorNotifiedImpl (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb-core/lib/cursor.js:561:38)
    at self._endSession (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb-core/lib/cursor.js:569:46)
    at ClientSession.endSession (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb-core/lib/sessions.js:129:41)
    at Cursor._endSession (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb-core/lib/cursor.js:189:13)
    at Cursor._endSession (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb/lib/cursor.js:226:59)
    at _setCursorNotifiedImpl (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb-core/lib/cursor.js:569:17)
    at setCursorNotified (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb-core/lib/cursor.js:561:3)
    at setCursorDeadAndNotified (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb-core/lib/cursor.js:554:3)
    at nextFunction (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb-core/lib/cursor.js:633:12)
    at Cursor.next (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb-core/lib/cursor.js:833:3)
    at Cursor._next (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb/lib/cursor.js:211:36)
    at fetchDocs (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb/lib/operations/cursor_ops.js:204:12)
    at cursor._next (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb/lib/operations/cursor_ops.js:230:7)
    at handleCallback (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb-core/lib/cursor.js:203:5)
    at nextFunction (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb-core/lib/cursor.js:748:5)
    at /home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb-core/lib/cursor.js:624:7
    at queryCallback (/home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb-core/lib/cursor.js:268:16)
    at /home/dumont_s/localbio/api/node_modules/loopback-connector-mongodb/node_modules/mongodb-core/lib/connection/pool.js:532:18
    at process._tickCallback (internal/process/next_tick.js:61:11)

```

#318 and #248 are similar issues with wrong data format saved in db, but seems to be caused by $set operator not supported in `toDatabase`, while this issue is caused by missing calls to `toDatabase`

**Comment by sylvaindumont** [comment]
It is still relevant, it should be fixed by #522

**Comment by sylvaindumont** [comment]
this issue is fixed by #522 but the fix is not released yet
